From 999b7f38923c57bc962848e7c708267830a171f7 Mon Sep 17 00:00:00 2001
From: Javen Ni <javen.ni@borqs.com>
Date: Fri, 26 Apr 2013 16:57:34 +0300
Subject: [PORT FROM MAIN] Telephony: Fix issues with cursor closing

BZ: 16858

There are unclosed cursor in telephony framework, which
creats issue with MMS downloading, in MTBF tests.

The fix was done by borqs in GB and later ported to ICS
and then to JB.

Change-Id: I9844efda3259030cb0380c2fa8db037b7c73d560
Orig-Change-Id: I1643b38917ae06b7d4a826e9f2a432added8e476
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Javen Ni <javen.ni@borqs.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/dataconnection/DcTracker.java        |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
index 9e6140f..a9f46ba 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
@@ -965,6 +965,10 @@ public final class DcTracker extends DcTrackerBase {
                 }
             } while (cursor.moveToNext());
         }
+        if (cursor != null) {
+            cursor.close();
+        }
+
         if (DBG) log("createApnList: X result=" + result);
         return result;
     }
-- 
1.7.4.1

